Lib/inspect.py

```diff
@@ -2276,10 +2276,22 @@ def _signature_fromstr(cls, obj, s, skip_bound_arg=True):
 
     def parse_name(node):
         assert isinstance(node, ast.arg)
-        if node.annotation is not None:
-            raise ValueError("Annotations are not currently supported")
         return node.arg
 
+    def parse_annotation(node):
+        assert isinstance(node, (ast.arg, ast.FunctionDef))
+        if isinstance(node, ast.arg):
+            annotation = node.annotation
+        else:
+            annotation = node.returns
+        if annotation:
+            expr = ast.unparse(annotation)
+            try:
+                return eval(expr, sys_module_dict)
```

Comment:
Shouldn't the eval logic here match wrap_value?

Reply:
Good catch.